nemo/core/classes/modelPT.py

```diff
@@ -421,6 +422,14 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
                 The list of "arg_value" will be parsed and a dictionary of optimizer kwargs \
                 will be built and supplied to instantiate the optimizer.
         """
+        if self._optimizer_param_groups is None:
+            logging.info(
```

Comment:
Why not just call self.setup_optimizer_param_groups() here instead of raising a warning and exiting ?
If it's None after calling that, directly raise error

Reply:
Plus this way setup param groups gets called by used just calling setup_optimization() in one go.